uci_ssh.go

```diff
@@ -0,0 +1,121 @@
+package uci
+
+import (
+	"bytes"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+)
+
+type SshTree struct {
+	host   string
+	config ssh.ClientConfig
+	client *ssh.Client
+	tree
+	sync.Mutex
+}
+
+func NewSshTree(username string, password string, host string) (t *SshTree) {
+	config := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(password),
+		},
+		// Non-production only
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         5 * time.Second,
+	}
```

Comment:
This is neither secure, nor very flexible. Let the user decide if they want to ignore host keys, use SSH certificates, restrict ciphers, etc.

```suggestion
func NewSshTree(config *ssh.ClientConfig, host string) (t *SshTree) {
```